CI: add compile tests

iomartin opened the pull request:
Make sure that the project compiles.

Runs on pushes and on pull requests

stefanhaRH committed:
gitmodules: use GitLab repos instead of qemu.org

qemu.org is running out of bandwidth and the QEMU project is moving
towards a gating CI on GitLab. Use the GitLab repos instead of qemu.org
(they will become mirrors).

Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>
Reviewed-by: Wainer dos Santos Moschetta <wainersm@redhat.com>
Reviewed-by: Thomas Huth <thuth@redhat.com>
Reviewed-by: Philippe Mathieu-Daudé <philmd@redhat.com>
Message-id: 20210111115017.156802-3-stefanha@redhat.com
Signed-off-by: Stefan Hajnoczi <stefanha@redhat.com>

iomartin committed:
CI: add compile tests

Make sure that the project compiles.

Runs on pull requests to the hermes branch

sbates130272 commented on .github/workflows/compile.yml:
```diff
@@ -0,0 +1,22 @@
+       - uses: actions/checkout@v2
+         with:
+           ref: ${{ github.event.pull_request.head.sha }}
+       - name: Install uBPF
```
Do we want to lock this to a tag of ubpf?

sbates130272 commented on .github/workflows/compile.yml:
```diff
@@ -0,0 +1,22 @@
+name: compile test
+
+on: [push, pull_request]
```
I don't think we want this on pushes to dev branches as they are often broken. I normally do this on PR against master and then also make it part of the master branch rules.

iomartin commented:
@sbates130272, I fixed the code to address your comments.

I was getting some timeouts when running the CI and it turns out it was a bandwidth issue on where some of the git modules are hosted. So I cherry-picked a commit from upstream that changes the address of those modules and I added to this PR as well.

sbates130272 reviewed (approved):
Thanks @iomartin. Looks good!
